nylas/models/drafts.py

```diff
@@ -0,0 +1,162 @@
+        bcc: The BCC recipients of the message.
+        reply_to: The reply-to recipients of the message.
+        unread: Whether the message is unread.
+        starred: Whether the message is starred.
+        snippet: A snippet of the message body.
+        body: The body of the message.
+        attachments: The attachments on the message.
+        send_at: Unix timestamp to send the message at.
+        reply_to_message_id: The ID of the message that you are replying to.
+        tracking_options: Options for tracking opens, links, and thread replies.
+    """
+
+    body: NotRequired[str]
+    thread_id: NotRequired[str]
+    subject: NotRequired[str]
+    snippet: NotRequired[str]
+    to: NotRequired[List[EmailName]]
+    bcc: NotRequired[List[EmailName]]
+    cc: NotRequired[List[EmailName]]
+    reply_to: NotRequired[List[EmailName]]
+    attachments: NotRequired[List[CreateAttachmentRequest]]
+    unread: NotRequired[bool]
+    starred: NotRequired[bool]
+    send_at: NotRequired[int]
+    reply_to_message_id: NotRequired[str]
+    tracking_options: NotRequired[TrackingOptions]
+
+
+UpdateDraftRequest = CreateDraftRequest
+""" A request to update a draft. """
+
+
+ListDraftsQueryParams = TypedDict(
+    "ListDraftsQueryParams",
+    {
+        **get_type_hints(ListQueryParams),
+        "subject": NotRequired[str],
+        "any_email": NotRequired[List[str]],
+        "from": NotRequired[List[str]],
+        "to": NotRequired[List[str]],
+        "cc": NotRequired[List[str]],
+        "bcc": NotRequired[List[str]],
+        "in": NotRequired[List[str]],
+        "unread": NotRequired[bool],
+        "starred": NotRequired[bool],
+        "thread_id": NotRequired[str],
+        "has_attachment": NotRequired[bool],
+    },
+)
+"""
+Query parameters for listing drafts.
+
+Attributes:
+    subject: Return messages with matching subject.
+    any_email: Return messages that have been sent or received by this comma-separated list of email addresses.
+    from: Return messages sent from this email address.
+    to: Return messages sent to this email address.
+    cc: Return messages cc'd to this email address.
+    bcc: Return messages bcc'd to this email address.
+    in: Return messages in this specific folder or label, specified by ID.
+    unread: Filter messages by unread status.
+    starred: Filter messages by starred status.
+    has_attachment: Filter messages by whether they have an attachment.
+    limit (NotRequired[int]): The maximum number of objects to return.
+        This field defaults to 50. The maximum allowed value is 200.
+    page_token (NotRequired[str]): An identifier that specifies which page of data to return.
+        This value should be taken from a ListResponse object's next_cursor parameter.
+"""
+
+
+class SendMessageRequest(CreateDraftRequest):
+    """
+    A request to send a message.
+
+    Attributes:
+        thread_id (NotRequired[str]): The thread that this message belongs to.
```

Comment:
do we want to consistently declare types in the docstrings? just noticing this is very inconsistent throughout the SDK

Reply:
So here's the thing - docstrings kinda suck for more complex applications, such as inheritance. If we don't mention the fields (and their types) of the superclass, it does not automatically inherit it in the docs, so in cases like this, we need to manually add it (and its types).

On top-level objects, all we need to do is mention the field name, and the docstring will be smart enough to take the type of the field from the code directly. This issue is strictly with sub-classes.